test/test_torch.py

```diff
@@ -16537,6 +16539,24 @@ def test_exp_slow(self, device, dtype):
         b = torch.exp(torch.ones(1, dtype=dtype, device=device))
         self.assertEqual(a, b.expand(2 ** 31))
 
+    def _unary_op_vs_numpy_template(self, torch_fn, np_fn, device, dtype, with_extremal):
```

Comment:
exp is currently tested in TestTorchMathOps, but that's being refactored into a new testing mechanism that uses "OpInfos". Take a look at how unary ufuncs like exp2 are registered here:

https://github.com/pytorch/pytorch/blob/6324ef4cedbaeea2799e69dab91c8e7cc6cc80f0/torch/testing/_internal/common_methods_invocations.py#L203

You can register exp2 the same way and it will be tested automatically. You'll need to fill out the `ref` field and set the `dtypesIfCPU` and `dtypesIfCUDA` fields since exp2 doesn't support complex values.

The OpInfo method of testing is new and I would appreciate your feedback trying to use it. I hope it's straightforward.

Reply:
Should I remove this test as the PR will update the `UnaryUfuncInfo` Operator list. Or both existing would fine. ( I think removal makes more sense, as both do similar testing)

Reply:
Yep. It should be redundant with the OpInfo-based tests (in test_unary_ufuncs.py). Better remove it.

Reply:
Had to do 
```python
UnaryUfuncInfo('exp2',
                   ref=np.exp2,
                   dtypes=floating_types_and(torch.half),
                   dtypesIfCPU=None, # Had to explicitly set to None
                   dtypesIfCUDA=None,), # Had to explicitly set to None
```

One would expect that if only `dtypes` is given all devices will verify only those dtypes based on the code below
https://github.com/pytorch/pytorch/blob/63ca5e096c8d1a814c6972d02c461017ee7d9243/torch/testing/_internal/common_methods_invocations.py#L73-L76

But with the following defaults in constructor,
https://github.com/pytorch/pytorch/blob/63ca5e096c8d1a814c6972d02c461017ee7d9243/torch/testing/_internal/common_methods_invocations.py#L160-L167

That behaviour is not followed. I think the default value for those arguments should be `None`.

Reply:
You're right that you have to explicitly override because of the defaults. We could add a more complex inference system for the default dtypes where the defaults are None but None is interpreted as first checking dtypes, if set, and then reverting to its current default value. Let's see if it comes up again?